### PR TITLE
fix: namespace properties to avoid key conflicts

### PR DESCRIPTION
**Pull request by timrbula**
## Context

Related Issues(s): Closes https://github.com/boomerang-io/roadmap/issues/335

Build Number: 3.7.1

## Checklist:

- [x] Verified in an integrated environment
- [x] Unit and integration tests passing
- [x] Semantic HTML with accessibility support
- [x] No linting, test console, or browser console errors (best effort)
- [x] JSDoc comment blocks for complex code

## PR Review Guidance

1. Update a parameter to a Workflow that matches a key of the schedule e.g. `name`, `id`, `dateTime`. 
2. Create/update a schedule and ensure you can update the `Workflow Parameters` section without any conflict

## Additional Info

I added transformations to the workflow parameter keys to use a `$property:` prefix to the parameter key e.g. `name` -> `$parameter:name`.

`$` and `:` aren't allowed characters for the parameter key value so I think it should be fine.

### CODE REVIEW

**Comment by timrbula** [comment]
@BenjaminRuby If you have a moment can you give this a quick review. I've verified the changes locally and w/ port forwarding and look to be working just fine

**Comment by BenjaminRuby** [comment]
Hey @timrbula I will review this one tomorrow - I still need to configure port forwarding setup on my new laptop.

Do you know why the build is failing in CICD? The same issues when building locally appear to me as warnings.

**Comment by timrbula** [comment]
@BenjaminRuby it's because of the warnings from the scss compilation. I have fixed those issues and pushed up some changes to this branch.

File: src/index.js [hunk 1]
```diff
@@ -1,35 +1,37 @@
...
-if (window.Cypress) {
-  new Server({
-    environment: "test",
-    routes() {
-      let methods = ["get", "put", "patch", "post", "delete"];
-      methods.forEach((method) => {
-        this[method]("/*", async (schema, request) => {
-          let [status, headers, body] = await window.handleFromCypress(request);
-          return new Response(status, headers, body);
+(async () => {
```
Comment (timrbula) [inline src/index.js:8]: This removes the json from the bundle. Definitely something we should look at for our other apps to make sure that we aren't including dev files in the final build unncessarily.

File: src/Features/Editor/Properties/PropertiesModal/WorkflowEditButton/EditButton.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import React from "react";
```
Comment (timrbula) [inline src/Features/Editor/Properties/PropertiesModal/WorkflowEditButton/EditButton.js:1]: Not ideal but ended up duplicating these files to fix the failing builds bc of the import order of the files. I couldn't figure out what was out of order.

File: src/Components/WorkflowTaskResultsForm/WorkflowTaskResultsForm.js [hunk 3]
```diff
@@ -12,7 +12,7 @@ import {
 import { Button, ModalBody, ModalFooter } from "@boomerang-io/carbon-addons-boomerang-react";
 import TextEditorModal from "Components/TextEditorModal";
 import { TEXT_AREA_TYPES } from "Constants/formInputTypes";
-import styles from "./WorkflowTaskForm.module.scss";
+import styles from "./WorkflowTaskFormResults.module.scss";
```
Comment (timrbula) [inline src/Components/WorkflowTaskResultsForm/WorkflowTaskResultsForm.js:15]: Renamed for clarity

File: src/Components/ScheduleManagerForm/ScheduleManagerForm.tsx [hunk 4]
```diff
@@ -63,9 +63,19 @@ export default function CreateEditForm(props: CreateEditFormProps) {
     workflow: props.workflow,
     days: [],
     labels: [],
-    ...props.schedule?.parameters,
   };
 
+  /**
```
Comment (timrbula) [inline src/Components/ScheduleManagerForm/ScheduleManagerForm.tsx:68]: These are the actual changes to fix the bug as noted in the PR description.

File: package.json [hunk 5]
```diff
@@ -120,11 +120,12 @@
     "lint-staged": "10.1.1",
     "miragejs": "^0.1.40",
     "ncp": "^2.0.0",
-    "node-sass": "^7.0.1",
+    "sass": "^1.50.0",
```
Comment (timrbula) [inline package.json:123]: sass is a drop in replacement for the deprecated `node-sass`

Comment (BenjaminRuby) [reply package.json:123]: another thing to look at replacing for the rest of our apps

**Review by BenjaminRuby — APPROVED** [review]
This appears to be working for me @timrbula 👍

**Pull request merged**